### PR TITLE
Implement scrollview interaction for load/save window

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Just committing the work I started last week for now. Doesn't look to difficult, but help is welcome due to time constraints.

### CODE REVIEW

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp [hunk 1]
```diff
@@ -275,6 +275,80 @@ namespace OpenLoco::Ui::PromptBrowse
...
+        auto index = y / self->row_height;
+        if (index > _numFiles)
+            return;
+
+        Audio::playSound(Audio::sound_id::click_down, self->x + (self->width / 2));
+
+        file_entry entry = _files[index];
+
+        // Clicking a directory, with left mouse button?
+        if (Input::state() == Input::input_state::scroll_left && entry.is_directory())
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/PromptBrowseWindow.cpp:290]: We use this check throughout, but I think this enum is either confusingly named, or wrong. This state appears to be just a normal left click, at any rate.

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/PromptBrowseWindow.cpp:290]: Its the state for a left click on a scroll area.

**Comment by AaronVanGeffen** [comment]
I'm hoping this will resolve #650.

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp [hunk 2]
```diff
@@ -775,26 +831,103 @@ namespace OpenLoco::Ui::PromptBrowse
...
+
+            // Does the file already exist?
+            if (fs::exists(path))
             {
-                registers regs;
-                regs.esi = (int32_t)window;
-                call(0x00446598, regs);
+                // Copy directory and filename to buffer.
+                char* buffer_2039 = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+                strncpy(&buffer_2039[0], path.c_str(), 512);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/PromptBrowseWindow.cpp:858]: Some of the CI jobs are complaining about going from `fs::value_type*` to `const char*` here, whilst others are fine. Any ideas how best to resolve this? I'd rather avoid casting, if possible.

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/PromptBrowseWindow.cpp:858]: Will be wchar_t on Windows and char on posix. You'll need to put it through some sort of conversion prior to this i think.

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp [hunk 3]
```diff
@@ -275,6 +277,60 @@ namespace OpenLoco::Ui::PromptBrowse
...
+        if (Input::state() == Input::input_state::scroll_left)
+        {
+            // Copy the selected filename without extension to text input buffer.
+            strncpy(_text_input_buffer, entry.get_name().data(), std::size(_text_input_buffer));
+
+            // Continue processing for load/save.
+            processFileForLoadSave(self);
+        }
+
+        // Clicking a file, with right mouse button
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/PromptBrowseWindow.cpp:309]: remove the white space here as its hard to follow otherwise.

**Review by duncanspumpkin — APPROVED** [review]

File: src/OpenLoco/Windows/PromptBrowseWindow.cpp (hunk 3 above)
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/PromptBrowseWindow.cpp:309]:
```suggestion
        // Clicking a file, with right mouse button
```
@duncanspumpkin Like this?

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/PromptBrowseWindow.cpp:309]: yes
